core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -544,7 +568,10 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
     val response = RequestKeyToResponseDeserializer(key).getMethod("parse", classOf[ByteBuffer]).invoke(null, resp).asInstanceOf[AbstractRequestResponse]
     val errorCode = RequestKeyToErrorCode(key).asInstanceOf[(AbstractRequestResponse) => Short](response)
 
-    val possibleErrorCodes = resources.map(_.errorCode)
+    var possibleErrorCodes = resources.map(_.errorCode)
+    if (key == ApiKeys.METADATA.id) 
+      possibleErrorCodes = possibleErrorCodes + Errors.UNKNOWN_TOPIC_OR_PARTITION.code()
```

Comment:
Looks like `ResourceType.errorCode` is lying. Not sure if that method makes sense, maybe we should remove it since it's only used by tests, but that needs further discussion so let's just ignore it for this PR.

Is `METADATA` the only case where we can also have `UNKNOWN_TOPIC_OR_PARTITION`? I thought there were more of them based on the code changes.